src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/cdk/cdk_stacks/tests/test_default_pipeline_type.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+# pylint: skip-file
+
+import pytest
+from pprint import pprint
+
+from aws_cdk import core
+from cdk_stacks.main import PipelineStack
+
+
+def test_pipeline_creation_outputs_as_expected_when_input_has_1_target_with_2_waves():
+    region_name = "eu-central-1"
+    acount_id = "123456789012"
+
+    stack_input = {
+        "input": {"params": {}, "default_providers": {}, "regions": {}},
+        "ssm_params": {"fake-region": {}},
+    }
+
+    stack_input["input"]["name"] = "test-stack"
+
+    stack_input["input"]["default_providers"]["source"] = {
+        "provider": "s3",
+        "properties": {"account_id": "123456789012"},
+    }
+    stack_input["input"]["default_providers"]["build"] = {
+        "provider": "codebuild",
+        "properties": {"account_id": "123456789012"},
+    }
+
+    stack_input["ssm_params"][region_name] = {
+        "modules": "fake-bucket-name",
+        "kms": f"arn:aws:kms:{region_name}:{acount_id}:key/my-unique-kms-key-id",
+    }
+    app = core.App()
+    PipelineStack(app, stack_input)
+
+    cloud_assembly = app.synth()
+    resources = {
+        k[0:-8]: v for k, v in cloud_assembly.stacks[0].template["Resources"].items()
+    }
+    code_pipeline = resources["codepipeline"]
+    assert code_pipeline["Type"] == "AWS::CodePipeline::Pipeline"
+    assert len(code_pipeline["Properties"]["Stages"]) == 2
+
+    source_stage = code_pipeline["Properties"]["Stages"][0]
+    assert len(source_stage["Actions"]) == 1
+
+    source_stage_action = source_stage["Actions"][0]
+    assert source_stage_action["ActionTypeId"]["Category"] == "Source"
+    assert source_stage_action["ActionTypeId"]["Owner"] == "AWS"
+    assert source_stage_action["ActionTypeId"]["Provider"] == "S3"
+
+    build_stage = code_pipeline["Properties"]["Stages"][1]
+    build_stage_action = build_stage["Actions"][0]
+    assert build_stage_action["ActionTypeId"]["Category"] == "Build"
+    assert build_stage_action["ActionTypeId"]["Owner"] == "AWS"
+    assert build_stage_action["ActionTypeId"]["Provider"] == "CodeBuild"
+
+    assert len(build_stage["Actions"]) == 1
+
+    region_name = "eu-central-1"
```

Comment:
It looks like another test kicks-off from this line.
Could you add its function/test definition?